src/components/api.js

```diff
@@ -0,0 +1,223 @@
+export default class Api {
+  constructor(options) {}
+
+  //methods
+  getInitialCards() {
+    return fetch("https://around-api.en.tripleten-services.com/v1/cards", {
+      headers: {
+        authorization: "4792ec92-cf1c-45a6-8740-0f5d63585faa",
+      },
+    })
+      .then((res) => {
+        if (res.ok) {
+          //check for successful response
+          console.log(res.ok);
+          return res.json();
+        }
+        //check for unsuccessful response
+        console.log("Error");
+        return Promise.reject(`getInitialCards method Error: ${res.status}`);
```

Comment:
```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the **reference** for the method rather than the call. 

**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.